src/Units/UnitModel.java

```diff
@@ -35,38 +35,62 @@
   public int maxFuel;
   public int idleFuelBurn;
   public int movePower;
+  public int vision;
+  public boolean piercingVision = false;
   public MoveType propulsion;
-  public ActionType[] possibleActions;
+  public ArrayList<ActionType> possibleActions = new ArrayList<ActionType>();
   public Set<TerrainType> healableHabs;
-  public WeaponModel[] weaponModels;
+  public ArrayList<WeaponModel> weaponModels = new ArrayList<WeaponModel>();
 
   public int maxHP;
   public int holdingCapacity;
   public Vector<UnitEnum> holdables;
   private int COstr;
   private int COdef;
 
-  public UnitModel(String pName, UnitEnum pType, ChassisEnum pChassis, int cost, int pFuelMax, int pIdleFuelBurn, int pMovePower, MoveType pPropulsion,
-      ActionType[] actions, WeaponModel[] weapons)
+  public UnitModel(String pName, UnitEnum pType, ChassisEnum pChassis, int cost, int pFuelMax, int pIdleFuelBurn, int pVision, int pMovePower,
+      MoveType pPropulsion, ActionType[] actions, WeaponModel[] weapons)
+  {
+    this(pName, pType, pChassis, cost, pFuelMax, pIdleFuelBurn, pVision, pMovePower, pPropulsion);
+
+    for( ActionType action : actions )
+    {
+      possibleActions.add(action);
+    }
+    for( WeaponModel action : weapons )
+    {
+      weaponModels.add(action);
+    }
+  }
+
+  public UnitModel(String pName, UnitEnum pType, ChassisEnum pChassis, int cost, int pFuelMax, int pIdleFuelBurn, int pVision, int pMovePower,
+      MoveType pPropulsion, ArrayList<ActionType> actions, ArrayList<WeaponModel> weapons)
+  {
+    this(pName, pType, pChassis, cost, pFuelMax, pIdleFuelBurn, pVision, pMovePower, pPropulsion);
+    possibleActions = actions;
+    weaponModels = weapons;
+  }
+
+  private UnitModel(String pName, UnitEnum pType, ChassisEnum pChassis, int cost, int pFuelMax, int pIdleFuelBurn, int pVision, int pMovePower,
```

Comment:
If we are going to have multiple constructors, I'd rather have one be the "real" one, and the others just `this` it if possible. The "real" one should naturally be the one with all the parameters, and other constructors can supply default/placeholder values for the ones they don't have.
This would have the added benefit of being able to remove the weapon argument when calling in from e.g. APC, instead of supplying the mildly-painful `new WeaponModel[0]`

Reply:
Hah, fair. How do you propose we deal with the desire (need?) to have both basic array and ArrayList arguments?
I don't know how to make compile-time ArrayLists, and .toList() doesn't seem to work for passing into a constructor that takes a basic array.